src/general/IWorldEffect.cs

```diff
@@ -53,3 +54,71 @@ public void OnTimePassed(double elapsed, double totalTimePassed)
         }
     }
 }
+
+public class GasProductionEffect : IWorldEffect
+{
+    [JsonProperty]
+    private GameWorld targetWorld;
+
+    public GasProductionEffect(GameWorld targetWorld)
+    {
+        this.targetWorld = targetWorld;
+    }
+
+    public void OnRegisterToWorld()
+    {
+    }
+
+    public void OnTimePassed(double elapsed, double totalTimePassed)
+    {
+        List<Compound> gasCompounds = SimulationParameters.Instance.GetGasCompounds();
```

Comment:
Will this list be changing? I'd look into doing this just once in OnRegisterToWorld

Reply:
I'd slightly prefer this being here, because after all this runs just once for each editor entry. And re-checking the gas compounds list each time would make this more compatible with loading mods after creating a save that add more compound types.